File: models/message.js

/* Importing the mongoose module. */
const mongoose = require("mongoose");


/* Creating a schema for the message model. */
const msgSchema = mongoose.Schema({

    content:{
        required: true,
        type: String
    },

    senderID: {
        required: true ,
        type: mongoose.ObjectId
    },

    receiverID: {
        required: true ,
        type: mongoose.ObjectId
    },

    roomID:{
        type: String,
    },
    
    createDate:{
        type: Date,
        default: Date.now,
    }


    
});


/* Creating a model for the message schema. */
const Message = mongoose.model("Message", msgSchema);


/* Exporting the Message model so that it can be used in other files. */
module.exports = Message;

File: lib/models/message.dart
import 'package:rentcarmobile/utils/message_type.dart';

class Message {
  var content = "null";
  //var messageType = MessageType.Sent;
  String messageType = "null";
  String createDate = DateTime.now().toString();
  String senderID = "null";
  String receiverID = "null";
  String roomID = "null";
  Message(this.content, this.messageType, this.createDate);
  Message.msg(this.content, this.messageType, this.createDate, this.senderID,
      this.receiverID, this.roomID);
}